dlc-oracle/src/main/scala/org/bitcoins/dlc/oracle/DLCOracle.scala

```diff
@@ -91,85 +91,207 @@ case class DLCOracle(private val extPrivateKey: ExtPrivateKeyHardened)(implicit
 
   def listPendingEventDbs(): Future[Vector[EventDb]] = eventDAO.getPendingEvents
 
-  def listEvents(): Future[Vector[Event]] = {
-    for {
-      rValDbs <- rValueDAO.findAll()
-      eventDbs <- eventDAO.findAll()
-      outcomes <- eventOutcomeDAO.findAll()
-    } yield {
-      val rValDbsByNonce = rValDbs.groupBy(_.nonce)
-      val outcomesByNonce = outcomes.groupBy(_.nonce)
-      eventDbs.map(db =>
-        Event(rValDbsByNonce(db.nonce).head, db, outcomesByNonce(db.nonce)))
+  def listEvents(): Future[Vector[OracleEvent]] = {
+    eventDAO.findAll().map { eventDbs =>
+      val events = eventDbs.groupBy(_.eventDescriptorTLV)
+      events.values.map(dbs => OracleEvent(dbs)).toVector
     }
   }
 
-  def getEvent(nonce: SchnorrNonce): Future[Option[Event]] = {
+  def getEvent(nonce: SchnorrNonce): Future[Option[OracleEvent]] = {
     for {
-      rValDbOpt <- rValueDAO.read(nonce)
       eventDbOpt <- eventDAO.read(nonce)
-      outcomes <- eventOutcomeDAO.findByNonce(nonce)
-    } yield {
-      (rValDbOpt, eventDbOpt) match {
-        case (Some(rValDb), Some(eventDb)) =>
-          Some(Event(rValDb, eventDb, outcomes))
-        case (None, None) | (Some(_), None) | (None, Some(_)) =>
-          None
+
+      oracleEventOpt <- eventDbOpt match {
+        case Some(eventDb) =>
+          getEvent(eventDb.eventDescriptorTLV)
+        case None =>
+          FutureUtil.none
       }
+    } yield oracleEventOpt
+  }
+
+  def getEvent(descriptor: EventDescriptorTLV): Future[Option[OracleEvent]] = {
+    eventDAO.findByDescriptor(descriptor).map { dbs =>
+      if (dbs.isEmpty) {
+        None
+      } else Some(OracleEvent(dbs))
     }
   }
 
-  def createNewEvent(
+  def createNewLargeRangedEvent(
+      eventName: String,
+      maturationTime: Instant,
+      base: UInt16,
+      isSigned: Boolean,
+      numDigits: Int): Future[OracleAnnouncementTLV] = {
+    require(base > UInt16.zero,
+            s"base cannot be less than 1, got ${base.toInt}")
+    require(numDigits > 0, s"numDigits cannot be less than 1, got $numDigits")
+
+    val descriptorTLV = if (isSigned) {
+      UninitializedLargeRangeEventDescriptorV0(base, isSigned, numDigits + 1)
```

Comment:
Why `+ 1` here, is this implicitly required because of number representation? https://github.com/bitcoin-s/bitcoin-s/pull/2185#discussion_r506661952

Reply:
If the number is signed then there is an extra nonce for signing if it is positive or negative.